drt_1d.py

```
@@ -32,47 +34,152 @@ def construct_source_term(num_voxels:int, source_voxel_index:int, source_strengt
+    - float: Streaming operator. This represents the probability that a particle born at ri streams to rj without interacting.
+    
+    Example Diagrams:
     
-    TODO: Update this function to accurately calculate tau for two arbitrary points within the voxelized space, considering the actual path of transport.
+      -->  Streaming to the right -->
+    ----------------------------------------------------
+    | Start Voxel  | In-between voxels |  End Voxel  | 
+    ----------------------------------------------------
+    |---[]---------|-------------------|---------[]--|
+    |   ri         |                   |         rj  |
+        |__________|                   |__________|
+         left distance                  right distance
+         
+                  
+      <--  Streaming to the left <--
+    ----------------------------------------------------
+    | End Voxel    | In-between voxels | Start Voxel | 
+    ----------------------------------------------------
+    |---[]---------|-------------------|---------[]--|
+    |   rj         |                   |         ri  |
+        |__________|                   |__________|
+         left distance                  right distance
     """
-    logging.debug("Calculating K_trans from voxel %d to voxel %d with voxel size %f", ri, rj, voxel_size)
+    logging.debug("Calculating K_trans from index %d to index %d, position %.3f to position %.3f, with voxel size %f", start_index, end_index, ri, rj, voxel_size)
     
     if voxel_size <= 0:
         raise ValueError("Voxel size must be positive and non-zero.")
-    if ri < 0 or ri >= len(sigma_s) or rj < 0 or rj >= len(sigma_s):
-        raise ValueError("Voxel indices must be non-negative and within the range of the sigma_s_matrix.")
+    if ri < 0 or ri > len(sigma_s) * voxel_size or rj < 0 or rj > len(sigma_s) * voxel_size:
+        raise ValueError("Positions must be within the spatial domain defined by the voxels.")
```

Do this before your debugging line above - fail early